pkg/prometheus/promcfg.go

```diff
@@ -712,6 +724,30 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "target_limit", Value: getLimit(m.Spec.TargetLimit, enforcedTargetLimit)})
 	}
 
+	if m.Spec.LabelLimit > 0 || enforcedLabelLimit != nil &&
+		(version.Major == 2 && version.Minor >= 27) {
+		cfg = append(cfg, yaml.MapItem{Key: "label_limit", Value: getLimit(m.Spec.LabelLimit, enforcedLabelLimit)})
+	} else {
+		level.Warn(cg.logger).Log("msg", "PodMonitor 'labelLimit' is only available from prometheus 2.27.",
+			"podMonitor", m.Name)
+	}
```

Comment:
You need to check the version in a nested `if` otherwise we will log the warning even when `labelLimit` isn't set.